website/src/content/docs/docs/libraries/http/files.md

```diff
@@ -0,0 +1,413 @@
+  @bodyRoot body: {
+    ...File;
+  },
+): void;
+```
+
+### Using files in multipart payloads
+
+Multipart payloads are commonly used to upload files (e.g. in HTML forms). To declare a multipart part that has a file body, ensure the part's type follows the same rules as for request and response payloads: it must either _be_ a type that is effectively an instance of `Http.File`, or must have an explicit body property that _effectively_ is-or-extends `Http.File`. All of the following examples declare multipart parts that have file bodies:
+
+```typespec
+// The type of the form-data part is _exactly_ a File, so the part has a file body.
+op multipartUpload(
+  @multipartBody fields: {
+    file: HttpPart<File>;
+  },
+): void;
+
+// The type of the form-data part has an explicit body that is _exactly_ a File, so the part has a file body.
+op multipartUpload(
+  @multipartBody fields: {
+    file: HttpPart<{
+      @bodyRoot file: File;
+    }>;
+  },
+): void;
+
+// The type of the mixed part is _exactly_ a File, so the part has a file body.
+op multipartMixedDownload(): {
+  @multipartBody data: [HttpPart<File>];
+};
+
+// The type of the mixed part has an explicit body that is _exactly_ a File, so the part has a file body.
+op multipartMixedDownload(): {
+  @multipartBody data: [
+    HttpPart<{
+      @bodyRoot file: File;
+    }>
+  ];
+};
+```
+
+All of the above examples will also have file bodies if `File` is replaced with a model that extends `File` or a model that is _effectively_ `File` (e.g. `{...File}`).
+
+You can also mix and match parts that have file bodies with other parts. The following TypeSpec gives a more comprehensive example of uploading data alongside files:
+
+```typespec
+model Widget {
+  id: string;
+  name: string;
+  weight: float64;
+}
+
+op multipartUpload(
+  @multipartBody fields: {
+    // The widget is uploaded in a part named `widget` and uses form-urlencoded serialization.
+    widget: HttpPart<{
+      @header contentType: "application/x-www-form-urlencoded";
+      @body widget: Widget;
+    }>;
+
+    // The part named `attachments` can be sent multiple times, and each `attachments` part has a file body.
+    attachments: HttpPart<File>[];
+  },
+): void;
+```
+
+For more information about the handling of multipart payloads in `@typespec/http`, see [Multipart](./multipart.md).
+
+### When a model is _effectively_ a `File`
+
+In the above sections, we used the idea of "effective" files. In the context of an HTTP operation, a model is _effectively_ a file if it has _all_ of the properties of `Http.File` (true properties of `Http.File` from a spread or intersection, not just properties that have the same shape as a `File`) **AND** after removing all of the _applicable_ metadata properties, it has _only_ properties of `Http.File`.
+
+- A property of `Http.File` means a property that is actually sourced from the `Http.File` model, e.g. through spreading `File` into another model or using `model is` syntax.
```

Comment:
> through spreading `File` into another model

There's some subtlety here.
This works:
`op create(): { ...File };`

But this would not be a file:
```
model FakeFile { ...File }
op create(): FakeFile;
```
You can spread File into a returnType/operation params, but not into another model and have that new model be treated as one.

Reply:
Merged your latest PR into this branch and resolving.